woocommerce/woocommerce-gutenberg-products-block
# Open Mini Cart drawer when adding to cart from the All Products block

Fixes #4688.

This PR includes several changes:
* When adding a product using the Add to Cart product element block, the events `wc-blocks_adding_to_cart` and `wc-blocks_added_to_cart` are now triggered. The Mini Cart block listens to those events, so it now opens when adding a product to cart from the All Products or Single Product blocks.
* I updated the code that triggers DOM events to create events with `CustomEvent`, that allows us to attach some data to them.
* The `wc-blocks_adding_to_cart` event now accepts an `invalidateCartData` property. If it's set to false, data in the cart data store will not be invalidated when triggered. Useful to avoid the Cart block reloading all its contents when a product is added to cart from a block that is already connected to our data store (ie All Products).
* I removed some IE11 specific code.
* I added a small check inside `showContents()` to load scripts if they hadn't been loaded yet. Should be useful if the `wc-blocks_added_to_cart` event is ever triggered without the `wc-blocks_added_to_cart` having been triggered before.

### Screenshots

https://user-images.githubusercontent.com/3616980/133733589-cf3b8d26-069e-4ac1-a910-59f4cf1df267.mp4

### Manual Testing

**Test that Mini Cart opens when adding a product to cart via the All Products block:**

1. Create a post or page with the Mini Cart block + the All Products block.
2. Add a product to cart using the All Products block.
3. Verify the drawer opens.

**Test that Cart block updates when adding products to cart via a PHP product grid block:**

1. Create a post or page with a PHP product grid block (ie: Handpicked products) and the Cart block.
2. Add a product to cart using the PHP product grid block.
3. Verify the Cart contents are reloaded.

**Test that Cart block updates when adding products to cart via the All Products block:**

1. Create a post or page with the All Products block and the Cart block.
2. Add a product to cart using the All Products block.
3. Verify the Cart contents are updated without causing the full cart to reload (there is no spinner covering the entire block, instead, the new product has just been automatically added).
